商品削除機能の実装

k-foo opened the pull request:
# What
- 商品削除機能を実装
- 出品者だけが商品情報を削除できるように実装（念のためビューファイルだけでなくコントローラー上でも出来ないように実装）
# Why
- 出品を削除することができるため
- 他の人の商品も消すことが出来ては困るため

<br>
<br>

【確認動画】
●ログインしている人に「削除ボタン」が表示される
https://gyazo.com/412d360b6f8461974562cdfe46249acf
●削除でき、トップページに戻る（空の場合にはダミー表示）
https://gyazo.com/dfd86a9433f0c1c30720460734de1b58
●データベース上でもなくなったことを確認
https://gyazo.com/f8c99e3b5aeaa9aacf0e0cd1e61d4848
●他の人の商品は削除できない（削除ボタンがない）
https://gyazo.com/4763cccbf74a639d0bd95702203fce11

k-foo committed:
商品削除機能の実装。

Akinori0123 commented on app/controllers/items_controller.rb:
```diff
@@ -44,6 +44,17 @@ def update
 
+  def destroy
+    # ログインしているユーザーと同一であればデータを削除する
+    item = Item.find(params[:id])
```
【修正任意（修正後のレビュー依頼は不要となります。修正された場合は動作確認のみお願いいたします）】
```
item = Item.find(params[:id])
```
6行目のset_itemを使えば49行目のitemの定義は不要になるかと考えられます（リファクタリング）。
itemが@itemになることだけ注意していただければと思います。

k-foo replied:
教えていただきありがとうございます！
リファクタリングしたいので修正しました。（動作も確認しました）

Akinori0123 reviewed (commented):
レビュー依頼ありがとうございます。
LGTMです。修正任意のコメントをしましたのでマージ前にご確認お願いいたします。
実装お疲れ様でした。引き続き頑張っていきましょう！

k-foo committed:
リファクタリング

ビフォーアクションを使えば、一行不要になるため、修正を行った。（削除自体はインスタンス変数でなくても削除できる。しかし、＠itemのインスタンス変数とすれば、消したいitemを消せる実装のまま、リファクタリングできる）
